game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -278,14 +278,17 @@ void addDependentUnits(final Map<Unit, Collection<Unit>> dependencies) {
 
   @Override
   public List<Unit> getRemainingDefendingUnits() {
-    final List<Unit> remaining = new ArrayList<>(defendingUnitsRetreated);
+    final Set<Unit> remaining = new HashSet<>(defendingUnitsRetreated);
+    remaining.addAll(defendingUnits);
```

Comment:
This is a bit of a surprising update. This line seems like a really important item. I'm now not sure how this was even close to correct before.